### PR TITLE
Rename lm to link_model

### PR DESCRIPTION
**Pull request by davetcoleman**
Super simple code cleanup via better variable name... I was debugging this part of the code.

### CODE REVIEW

File: moveit_core/robot_state/include/moveit/robot_state/robot_state.h [hunk 1]
```diff
@@ -1592,10 +1592,10 @@ as the new values that correspond to the group */
   void getAttachedBodies(std::vector<const AttachedBody*>& attached_bodies) const;
 
   /** \brief Get all bodies attached to a particular group the model corresponding to this state */
-  void getAttachedBodies(std::vector<const AttachedBody*>& attached_bodies, const JointModelGroup* lm) const;
+  void getAttachedBodies(std::vector<const AttachedBody*>& attached_bodies, const JointModelGroup* link_model) const;
```
Comment (v4hn) [inline moveit_core/robot_state/include/moveit/robot_state/robot_state.h]: a `JointModelGroup` is no link_model

Comment (rhaschke) [reply moveit_core/robot_state/include/moveit/robot_state/robot_state.h]: Fixed in new commit.

**Comment by v4hn** [comment]
> Fixed in new commit.

We just talked about maintainers cleaning up minor issues on review. *sigh*
I should have cleaned it up instead of adding a comment, sorry.

**Comment by davetcoleman** [comment]
Thanks for fixing my issue, much appreciated!

**Pull request merged**